modules/context/api.go

```diff
@@ -259,3 +259,61 @@ func (ctx *APIContext) NotFound(objs ...interface{}) {
 		"errors":            errors,
 	})
 }
+
+// RepoRefForAPI handles repository reference names when the ref name is not explicitly given
+func RepoRefForAPI() macaron.Handler {
```

Comment:
Maybe it's not necessary to double this function, just check whether it's api in SerrError and do different response is enough.
for example:

```DIFF
diff --git a/modules/context/context.go b/modules/context/context.go
index 1405860e0..d25889e0d 100644
--- a/modules/context/context.go
+++ b/modules/context/context.go
@@ -175,13 +175,29 @@ func (ctx *Context) ServerError(title string, err error) {
 }
 
 func (ctx *Context) serverErrorInternal(title string, err error) {
+	isAPIPath := auth.IsAPIPath(ctx.Req.URL.Path)
+
 	if err != nil {
 		log.ErrorWithSkip(2, "%s: %v", title, err)
-		if macaron.Env != macaron.PROD {
+		if macaron.Env != macaron.PROD && !isAPIPath {
 			ctx.Data["ErrorMsg"] = err
 		}
 	}
 
+	if isAPIPath {
+		message := ""
+		if err != nil && (macaron.Env != macaron.PROD || (ctx.User != nil && ctx.User.IsAdmin)) {
+			message = err.Error()
+		}
+
+		ctx.JSON(http.StatusInternalServerError, APIError{
+			Message: message,
+			URL:     setting.API.SwaggerURL,
+		})
+
+		return
+	}
+
 	ctx.Data["Title"] = "Internal Server Error"
 	ctx.HTML(http.StatusInternalServerError, base.TplName("status/500"))
 }
```

Reply:
@a1012112796 your suggestion is nice only in the first look ... because:

It would add complexity and merge normal Context and APIContext ... so if I apply your diff I just have added more code but didn't reduce code :/

Reply:
for more info just look at modules/context/api.go

Reply:
@a1012112796 if you rever to https://github.com/go-gitea/gitea/pull/13511#discussion_r520981375

some background info: I just tryed to remove that dependency first but found out, that a smimilar function was used all over the code to do the same ... so creating a new one and use this explicit for api is **NOW** fine